src/components/Shared/RecordRemarks.js

```diff
@@ -0,0 +1,162 @@
+import React, { useContext, useState } from 'react'
+import Grid from '@mui/system/Unstable_Grid/Grid'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { Box, IconButton, TableBody, Table, TableCell, TableContainer, TableRow } from '@mui/material'
+import Icon from 'src/@core/components/icon'
+import moment from 'moment'
+
+// import Table from 'src/components/Shared/Table'
+
+import { useResourceQuery } from 'src/hooks/resource'
+import RecordRemarksForm from './RecordRemarksForm'
+import { useWindow } from 'src/windows'
+import DeleteDialog from './DeleteDialog'
+
+const RecordRemarks = ({ recordId, resourceId, expanded }) => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { stack } = useWindow()
+  const [deleteDialogOpen, setDeleteDialogOpen] = useState([false, {}])
+  const userId = JSON.parse(window.sessionStorage.getItem('userData'))?.userId
+
+  const fetchGridData = () => {
+    var parameters = `_resourceId=${resourceId}&_masterRef=${recordId}`
+
+    return getRequest({
+      extension: SystemRepository.RecordRemarks.qry,
+      parameters: parameters
+    })
+  }
+
+  const date = date => {
+    const eventDate = moment(date)
+    const formattedDate = eventDate.calendar()
+
+    return formattedDate
+  }
+
+  const {
+    query: { data },
+    labels: labels,
+    access
+  } = useResourceQuery({
+    enabled: !!recordId,
```

Comment:
what is this used for ?

Reply:
to make make useQuery enabled where recordId ready to prevent request  error